data/shared/src/main/scala/org/ergoplatform/validation/ValidationRules.scala

```diff
@@ -162,6 +162,44 @@ object ValidationRules {
     override protected def settings: SigmaValidationSettings = currentSettings
   }
 
+  object CheckV6Type extends ValidationRule(1019,
+    "Check the type has the declared method.") {
+    override protected lazy val settings: SigmaValidationSettings = currentSettings
+
+    final def apply[T](v: EvaluatedValue[_]): Unit = {
+      checkRule()
+
+      def v6TypeCheck(tpe: SType) = {
+        if (tpe.isOption || tpe.typeCode == SHeader.typeCode || tpe.typeCode == SUnsignedBigInt.typeCode) {
+          throwValidationException(
+            new SerializerException(s"V6 type used in register or context var extension: $tpe"),
+            Array[Any](tpe))
+        }
+      }
+      v match {
+        case c: Constant[_] => v6TypeCheck(c.tpe)
+        case t: Tuple => if(t.items.length != 2) {
```

Comment:
It's possible to parse tuples with length > 2 although they don't seem to be very useful since they can't be evaluated, but they could be stuffed in register/contextextension

Reply:
addressed, no length checked now